lib/srv/alpnproxy/conn.go

```diff
@@ -171,8 +171,8 @@ func (c *PingConn) Write(p []byte) (int, error) {
 	// Avoid overflow when casting data length. It is only present to avoid
 	// panicking if the size cannot be cast. Callers should handle packet length
 	// limits, such as protocol implementations and audits.
-	if len(p) > math.MaxUint32 {
-		return 0, trace.BadParameter("invalid content size, max size permitted is %d", math.MaxUint32)
+	if uint64(len(p)) > math.MaxUint32 {
+		return 0, trace.BadParameter("invalid content size, max size permitted is %d", uint64(math.MaxUint32))
```

Comment:
nit: Is the cast necessary?

```suggestion
		return 0, trace.BadParameter("invalid content size, max size permitted is %d", math.MaxUint32)
```

Reply:
Yeah, it complains the same way as the line above. I think it is because `%d` considers the input as `int`.

Reply:
Oh, right. Use %v and drop the cast?

Reply:
It still complains, it might be because this constant is untyped and is being cast by default to `int`.